nixos/modules/services/security/sshguard.nix

```diff
@@ -104,22 +107,24 @@ in {
       after = [ "network.target" ];
       partOf = optional config.networking.firewall.enable "firewall.service";
 
-      path = with pkgs; [ iptables ipset iproute systemd ];
+      path = with pkgs; if config.networking.nftables.enable
+        then [ nftables iproute systemd ]
+        else [ iptables ipset iproute systemd ];
 
       # The sshguard ipsets must exist before we invoke
       # iptables. sshguard creates the ipsets after startup if
       # necessary, but if we let sshguard do it, we can't reliably add
       # the iptables rules because postStart races with the creation
       # of the ipsets. So instead, we create both the ipsets and
       # firewall rules before sshguard starts.
-      preStart = ''
+      preStart = optionalString config.networking.firewall.enable ''
```

Comment:
This change looks unrelated? Or is it required?

Reply:
It is related and required. `iptables` and `ipset` are not installed when `nftables` is used. The preStart script wouldn't work or make sense as it is not needed to prepare any ipsets or iptables rules.

Reply:
Makes perfect sense to me.